rbc/tests/test_omnisci_array_functions.py

```diff
@@ -0,0 +1,143 @@
+        'CREATE TABLE IF NOT EXISTS {table_name} ({table_defn});'
+        .format(**locals()))
+
+    def row_value(row, col, colname):
+        if colname == 'b':
+            return 'ARRAY[%s]' % (', '.join(
+                ("'true'" if i % 2 == 0 else "'false'")
+                for i in range(-3, 3)))
+        if colname.startswith('f'):
+            return 'ARRAY[%s]' % (', '.join(
+                str(row * 10 + i + 0.5) for i in range(-3, 3)))
+        return 'ARRAY[%s]' % (', '.join(
+            str(row * 10 + i) for i in range(-3, 3)))
+
+    rows = 5
+    for i in range(rows):
+        table_row = ', '.join(str(row_value(i, j, n))
+                              for j, n in enumerate(colnames))
+        m.sql_execute(
+            'INSERT INTO {table_name} VALUES ({table_row})'.format(**locals()))
+    m.table_name = table_name
+    yield m
+    try:
+        m.sql_execute('DROP TABLE IF EXISTS {table_name}'.format(**locals()))
+    except Exception as msg:
+        print('%s in deardown' % (type(msg)))
+
+
+def np_ones(sz):
+    return omni.ones(sz, types.int32)
+
+
+def np_ones_dtype(sz):
+    return omni.ones(sz)
+
+
+def np_ones_like_dtype(i4):
+    return omni.ones_like(i4, dtype=types.double)
+
+
+def np_ones_like(i4):
+    return omni.ones_like(i4)
+
+
+def np_zeros(sz):
+    return omni.zeros(sz, types.int32)
+
+
+def np_zeros_dtype(sz):
+    return omni.zeros(sz)
+
+
+def np_zeros_like(i4):
+    return omni.zeros_like(i4)
+
+
+def np_zeros_like_dtype(i4):
+    return omni.zeros_like(i4, dtype=types.double)
+
+
+def np_full(sz, fill_value):
+    a = omni.full(sz, fill_value, types.double)
+    return a
+
+
+def np_full_dtype(sz, fill_value):
+    a = omni.full(sz, fill_value)
+    return a
+
+
+def np_full_like(i1, fill_value):
+    return omni.full_like(i1, fill_value)
+
+
+def np_full_like_dtype(i1, fill_value):
+    return omni.full_like(i1, fill_value, dtype=types.double)
+
+
+def np_cumsum(sz):
+    a = omni.ones(5)
+    return np.cumsum(a)
+
+
+array_methods = [
+    ('full', 'double[](int64, double)', (5, 3), np.full(5, 3, dtype=np.int32)),
+    ('full_dtype', 'double[](int64, double)', (5, 3), np.full(5, 3)),
+    ('full_like', 'int8[](int8[], int32)', ('i1', 3), np.full(6, 3, dtype='b')),  # noqa: E501
+    ('full_like_dtype', 'double[](int8[], double)', ('i1', 3.0), np.full(6, 3, dtype='q')),  # noqa: E501
+    ('ones', 'int32[](int64)', (5,), np.ones(5, dtype=np.int32)),
+    ('ones_dtype', 'double[](int64)', (5,), np.ones(5)),
+    ('ones_like', 'int32[](int32[])', ('i4',), np.ones(6, dtype='i')),
+    ('ones_like_dtype', 'double[](int32[])', ('i4',), np.ones(6, dtype='q')),
+    ('zeros', 'int32[](int64)', (5,), np.zeros(5, dtype=np.int32)),
+    ('zeros_like', 'int32[](int32[])', ('i4',), np.zeros(6, dtype='i')),
+    ('zeros_like_dtype', 'double[](int32[])', ('i4',), np.zeros(6, dtype='q')),
+    ('zeros_dtype', 'double[](int64)', (5,), np.zeros(5)),
+    ('cumsum', 'double[](int32)', (5,), np.arange(1, 6)),
```

Comment:
The `cumsum` crashes omniscidb server (master version), gdb shows:
```
Thread 34 "omnisci_server" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffb1ffb700 (LWP 28468)]
0x00007fffe60c52b9 in rbc::omnisci_array::omnisci_np_cumsum::$3clocals$3e::impl$247($28STRUCT__dPLbK$29$2a) ()
(gdb) bt
#0  0x00007fffe60c52b9 in rbc::omnisci_array::omnisci_np_cumsum::$3clocals$3e::impl$247($28STRUCT__dPLbK$29$2a) ()
#1  0x0000000000000000 in ?? ()

```

Reply:
I can reproduce this error on omniscidb master

Reply:
The error happens because this test is calling `np.ones` with the constant value 5 rather than using the `sz` argument. When using the literal value 5 as input, the code is generated and later deleted by the LLVM DCE pass:

```llvm
; Function Attrs: noreturn nounwind
define void @np_cumsum____dPLbKPaiA({ double*, i64, i8 }* nocapture readnone %.1, i32 %.2) local_unnamed_addr #0 {
entry.endif:
  tail call void @llvm.trap()
  unreachable
}
```

Using a non-literal value seems to prevent DCE from deleting everything. I will open an issue to investigate this.

Reply:
xref: #109